### PR TITLE
OP-339 resolve hard coded paths for configurations files

### PR DESCRIPTION
**Pull request by mwithi**
See OP-339

Since configuration files (.properties) are expected in the classpath, they've been added to the `test/resources`

### CODE REVIEW

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+package org.isf.generaldata;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/generaldata/ConfigurationProperties.java:1]: Missing the license header

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			if (exitOnFail) System.exit(1);
+		}
+		return prop;
+	}
+	
+	/**
+	 * Method to retrieve a property
+	 * 
+	 * @param property
+	 * @param defaultValue
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/generaldata/ConfigurationProperties.java:41]: This doesn't appear in the method signature

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/generaldata/ConfigurationProperties.java:22]: Thanks!

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java (hunk 2 above)
Comment (mwithi) [reply src/main/java/org/isf/generaldata/ConfigurationProperties.java]: Thanks

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	 * 
+	 * @param property
+	 * @return
+	 */
+	protected String myGetProperty(String property) {
+		return prop.getProperty(property);
+	}
+	
+	/**
+     * Method to retrieve an integer property
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/generaldata/ConfigurationProperties.java:69]: alignment

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		try {
+			value = prop.getProperty(property).equalsIgnoreCase("YES");
+		} catch (Exception e) {
+			return defaultValue;
+		}
+		return value;
+	}
+	
+	/**
+     * Method to retrieve an double property
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/generaldata/ConfigurationProperties.java:104]: alignment here too

File: src/main/java/org/isf/generaldata/ExaminationParameters.java [hunk 5]
```diff
@@ -21,21 +21,15 @@
...
  * Settings for Examination form
  *
  * @author Mwithi
  */
-public class ExaminationParameters {
+public class ExaminationParameters extends ConfigurationProperties {
 	
-	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+	private static final String FILE_PROPERTIES = "examination.properties";
+	private final static boolean EXIT_ON_FAIL = false;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/generaldata/ExaminationParameters.java:32]: The typical order of modifiers is `private static final <datatype> <variableName>`;  all the EXIT_ON_FAIL definitions across the files are the same (obviously cut-n-paste).

Why isn't the field defined once in the base class (`ConfigurationProperties`) and set to `false` as there is only one occurrence that is set to `true` (in `Version.java`)?

Comment (mwithi) [reply src/main/java/org/isf/generaldata/ExaminationParameters.java]: > The typical order of modifiers is `private static final <datatype> <variableName>`; all the EXIT_ON_FAIL definitions across the files are the same (obviously cut-n-paste).

not exactly (it was copy-and-paste 😝) thanks!

> Why isn't the field defined once in the base class (ConfigurationProperties) and set to false as there is only one occurrence that is set to true (in Version.java)?

It was easier using the same pattern (still copy-and-paste) and was making clear the behavior for the single configuration, but thanks for the catch, I've modified now

File: src/main/java/org/isf/generaldata/ConfigurationProperties.java (hunk 4 above)
Comment (mwithi) [reply src/main/java/org/isf/generaldata/ConfigurationProperties.java]: Thanks!

File: src/main/java/org/isf/generaldata/SmsParameters.java [hunk 6]
```diff
@@ -21,16 +21,8 @@
...
-import java.util.Properties;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-public class SmsParameters {
+public class SmsParameters extends ConfigurationProperties {
 	
-	private final Logger logger = LoggerFactory.getLogger(this.getClass());
 	private static final String FILE_PROPERTIES = "sms.properties";
```
Comment (emecas) [inline src/main/java/org/isf/generaldata/SmsParameters.java:26]: another hard coded file name

File: src/main/java/org/isf/generaldata/TxtPrinter.java [hunk 7]
```diff
@@ -21,16 +21,8 @@
...
-import java.util.Properties;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-public class TxtPrinter {
+public class TxtPrinter extends ConfigurationProperties {
 	
-	private final Logger logger = LoggerFactory.getLogger(this.getClass());
 	private static final String FILE_PROPERTIES = "txtPrinter.properties";
```
Comment (emecas) [inline src/main/java/org/isf/generaldata/TxtPrinter.java:26]: same here

File: src/main/java/org/isf/generaldata/Version.java [hunk 8]
```diff
@@ -21,62 +21,31 @@
...
-import java.util.Properties;
+public class Version extends ConfigurationProperties {
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-public class Version {
-	
-	private final Logger logger = LoggerFactory.getLogger(this.getClass());
 	private static final String FILE_PROPERTIES = "version.properties";
```
Comment (emecas) [inline src/main/java/org/isf/generaldata/Version.java:26]: one more

File: src/main/java/org/isf/generaldata/XmppData.java [hunk 9]
```diff
@@ -19,34 +19,29 @@
...
+package org.isf.generaldata;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.util.Properties;
-
-public class XmppData {
+public class XmppData extends ConfigurationProperties {
+	
+	private static final String FILE_PROPERTIES = "xmpp.properties";
```
Comment (emecas) [inline src/main/java/org/isf/generaldata/XmppData.java:26]: one more

File: src/main/java/org/isf/sms/providers/GSMParameters.java [hunk 10]
```diff
@@ -21,16 +21,12 @@
...
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.isf.generaldata.ConfigurationProperties;
 
-public class GSMParameters {
+public class GSMParameters extends ConfigurationProperties {
 	
-	private final Logger logger = LoggerFactory.getLogger(this.getClass());
 	private static final String FILE_PROPERTIES = "GSM.properties";
```
Comment (emecas) [inline src/main/java/org/isf/sms/providers/GSMParameters.java:30]: one more

File: src/main/java/org/isf/sms/providers/SkebbyParameters.java [hunk 11]
```diff
@@ -21,19 +21,13 @@
...
+import org.isf.generaldata.ConfigurationProperties;
 
 /**
  * @author Mwithi
  */
-public class SkebbyParameters {
+public class SkebbyParameters extends ConfigurationProperties {
 	
-	private final Logger logger = LoggerFactory.getLogger(this.getClass());
 	private static final String FILE_PROPERTIES = "Skebby.properties";
```
Comment (emecas) [inline src/main/java/org/isf/sms/providers/SkebbyParameters.java:31]: one more and why capital ?

File: src/main/java/org/isf/utils/sms/SetupGSM.java [hunk 12]
```diff
@@ -46,8 +48,10 @@
 public class SetupGSM extends JFrame implements SerialPortEventListener {
 
 	private static final long serialVersionUID = 1L;
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+	private static final String FILE_PROPERTIES = "GSM.properties";
```
Comment (emecas) [inline src/main/java/org/isf/utils/sms/SetupGSM.java:52]: one more, CAPITAL

**Review by emecas — COMMENTED** [review]
I'm wondering if all those file names could be centralized in a single configuration file, or the abstraction level desired with hard coded file names per module is enough for the purpose of this PR?

File: src/main/java/org/isf/dicom/manager/DicomManagerFactory.java [hunk 13]
```diff
@@ -41,6 +42,8 @@
  */
 public class DicomManagerFactory {
 
+	private static final String FILE_PROPERTIES = "dicom.properties";
```
[the remark this answers is not in the document]
Comment (mwithi) [reply src/main/java/org/isf/dicom/manager/DicomManagerFactory.java:45]: I don't know how the code could "guess" the filename

File: src/main/java/org/isf/utils/sms/SetupGSM.java (hunk 12 above)
Comment (mwithi) [reply src/main/java/org/isf/utils/sms/SetupGSM.java:52]: is it clear that are physical configuration files with case also?

**Comment by mwithi** [comment]
I'm changing the scope of this PR and related issue OP-339 to "configuration files only" so we can merge.

For resourcers I've open a new issue OP-491

**Comment by mwithi** [comment]
@mizzioisf has approved via comment https://github.com/informatici/openhospital-gui/pull/447

**Pull request merged**